src/emitMixin.ts

```diff
@@ -1,10 +1,11 @@
-import { getCurrentInstance } from 'vue'
-
-export const createEmitMixin = () => {
-  const events: Record<string, unknown[]> = {}
+import { getCurrentInstance, App } from 'vue'
 
+export const attachEventListener = (vm: App) => {
   const emitMixin = {
     beforeCreate() {
+      let events: Record<string, unknown[]> = {}
+      this.__emitted = events
```

Comment:
How come we are doing this by attaching `__emitted` to the component instance instead of just returning the object and passing it to `mount` (like we did before?) Does this change/fix some behavior or just personal preference?

Imo it is better to avoid mutating the component instance - the less we mix VTU implementation details with the actual components, the better.

Reply:
To be able to track each Vue Instance's events, I had to create the `events` variable for each instance's scope. The way it was made before would not have worked.

I know it is hacky, I also did not love the idea, but I could not figure a different way to track the events for each instance. 

Since `findComponent` can now return arbitrary Component Instance, it made sense to do so.

Reply:
Yep, figured that it was something like that, no problem